Source/AI/navmesh.cpp

```diff
@@ -566,34 +566,38 @@ void NavMesh::Save(const string& level_name, const Path& level_path) {
 
     string nav_path = GenerateParallelPath("Data/Levels", "Data/LevelNavmeshes", ".nav", level_path);
 
-    if (config["allow_game_dir_save"].toBool() == false) {
+    if (config["allow_game_dir_save"].toBool() == false) {  // change save location to user directory
         nav_path = AssemblePath(GetWritePath(level_path.GetModsource()), nav_path);
     }
 
-    char zip_path[kPathSize];
+    char zip_path[kPathSize];  // Path of the outermost file
     FormatString(zip_path, kPathSize, "%s.zip", nav_path.c_str());
 
-    char in_zip_file_name[kPathSize];
-    FormatString(in_zip_file_name, kPathSize, "%s.zip", level_name.c_str());
+    char in_zip_file_name[kPathSize];                                         // Name of the file inside the outermost zip file
+    FormatString(in_zip_file_name, kPathSize, "%s.nav", level_name.c_str());  // used to be .zip
```

Comment:
is the engine still compatible with the old cached data after this change?

the old data has an internal filename called `.zip`, and the loader will either have to either look for both the old and new filename, or the data will need to be repackaged in order to work.

repackaging might also cause problems for mods that may have already shipped with a navmesh zip file, unless the backup behavior is to rebuild and replace the zip. in which case this will cause a load time regression on those mods.

Reply:
From the code that I saw, the unzipping process is filename independent; it reads the first (and only) entry in the zip file and returns the data associated with it, regardless of it's file extension or name, and is saved to the designated location with the correct extension. If more than one entry is detected in the zip archive, an error is logged. I don't think there should be conflicts; I generated a set of archives using my custom fork (saved as `.nav` extension) and was able to load them using the normal overgrowth branch (which expects `.zip` extension).

This was more to make it clear as to what type of file it actually is; at first I though the issue was that the internal zip file was corrupted, but it's actually an entirely different filetype with the `.nav` extension. My intention was to avoid this confusion in the future, since in my testing changing the file extension had no effect.

Reply:
Awesome. I was also initially confused about the internal ".zip" file that didn't open in 7zip. If it loads it in the normal branch and doesn't just give up and rebuild it, then then renaming this sounds like a very good idea.